### PR TITLE
Story CORE-481 | docs yamls

### PR DESCRIPTION
**Pull request by nicholaspcr**
Kind: story

### CODE REVIEW

File: docs/yamls/index.md [hunk 1]
```diff
@@ -23,6 +23,12 @@ In this way, when the file is written in one of the formats described below it c
 
 [definitions and examples](type.md)
 
+## Aliases
+> Responsible for defining an alternative name for a channel, useful when you want to make a channel's name distinct in a single dApp or set of dApps
```
Comment (franciscobonand) [inline docs/yamls/index.md]: "channel" -> "Channel"

File: docs/yamls/alias.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| apiVersion        | Specify what version of the API to use, for example `"v1"`                                                                                                                                                   |
+| kind              | Specifies which structure the file represents, in this case it would be `alias`                                                                                                                              |
+| meta              | Metadata of Type                                                                                                                                                                                             |
+| &rarr;name        | Type Name of the Alias                                                                                                                                                                                       |
```
Comment (franciscobonand) [inline docs/yamls/alias.md]: "Type Name of the Alias"?  
Also, you should point out that the name of the alias is something like "childDApp.childDAppBoundary"

File: docs/yamls/alias.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+| &rarr;reference   | String that is utilized to define certain tags to the Alias in question, a way for the user to categorize the numerous Alias in the cluster.                                                                 |
+| &rarr;Annotations | Definitions that can describe characteristics of the Alias that later on can be used to process/group the Alias in your cluster.                                                                             |
```
Comment (franciscobonand) [inline docs/yamls/alias.md]: "Annotations" shouldn't be capitalized

File: docs/yamls/alias.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+| &rarr;parent      | Defines the Alias context in the cluster through the path of the dApp in which it is stored, for example: `app1.app2` means that the Alias is defined in the `app2` and that this DApp is a child of `app1`. |
+|                   |
+| &rarr;UUID        | String that is a unique ID,, it is filled automatically when creating certain components. By default leave it empty                                                                                          |
+| Target            | specifies the name of the Channel in which this Alias points to.                                                                                                                                             |
```
Comment (franciscobonand) [inline docs/yamls/alias.md]: "Target" shouldn't be capitalized

File: docs/yamls/alias.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+| &rarr;name        | Type Name of the Alias                                                                                                                                                                                       |
+| &rarr;reference   | String that is utilized to define certain tags to the Alias in question, a way for the user to categorize the numerous Alias in the cluster.                                                                 |
+| &rarr;Annotations | Definitions that can describe characteristics of the Alias that later on can be used to process/group the Alias in your cluster.                                                                             |
+| &rarr;parent      | Defines the Alias context in the cluster through the path of the dApp in which it is stored, for example: `app1.app2` means that the Alias is defined in the `app2` and that this DApp is a child of `app1`. |
+|                   |
+| &rarr;UUID        | String that is a unique ID,, it is filled automatically when creating certain components. By default leave it empty                                                                                          |
```
Comment (franciscobonand) [inline docs/yamls/alias.md]: As this is the YAML definition for aliases, I believe there's no need for UUID here as the user will never set it's value

File: docs/yamls/alias.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+| Target            | specifies the name of the Channel in which this Alias points to.                                                                                                                                             |
+
+
+### YAML example
+```yaml
+kind: alias
+apiVersion: v1
+meta:
+  name: aliasOne
```
Comment (franciscobonand) [inline docs/yamls/alias.md]: This naming is wrong

File: docs/yamls/channel.md [hunk 7]
```diff
@@ -12,9 +12,9 @@
 | &rarr; reference  | String that is utilized to define certain tags to the Channel in question, a way for the user to categorize the numerous Channels in the cluster.                        |
 | &rarr;Annotations | Definitions that can describe characteristics of the Channel that later on can be used to process/group the Channels in your cluster.                                    |
 | &rarr; parent     | Defines the Channel context in the cluster through the path of the dApp in which it is stored, for example: `app1.app2` means that the Channel is defined in the `app2`. |
-| &rarr; sha256     | Tags images with their sha256 digest.                                                                                                                                    |
+| &rarr; uuid       | Universally unique identifier for the Channel.                                                                                                                           |
```
Comment (franciscobonand) [inline docs/yamls/channel.md]: As this is the YAML definition for channels, I believe there's no need for UUID here as the user will never set it's value

File: docs/yamls/dapp.md [hunk 8]
```diff
@@ -11,25 +11,31 @@
 | &rarr;reference      | String that is utilized to define certain tags to the dApp in question, a way for the user to categorize the numerous dApps in the cluster.                                                 |
 | &rarr;Annotations    | Definitions that can describe characteristics of the dApp that later on can be used to process/group the dApps in your cluster.                                                             |
 | &rarr;parent         | Defines dApp context in relation to the cluster. For example `app1.app2` would mean that this dApp is located on the path `root->app1->app2->app-name`. It is injected by the Inspr daemon. |
-| &rarr;sha256         | Tags images with their sha256 digest.                                                                                                                                                       |
+| &rarr;uuid           | Universally unique identifier for the DApp.                                                                                                                                                 |
```
Comment (franciscobonand) [inline docs/yamls/dapp.md]: As this is the YAML definition for dApps/Nodes, I believe there's no need for UUID here as the user will never set it's value

File: docs/yamls/dapp.md [hunk 9]
```diff
@@ -11,25 +11,31 @@
...
 | reference            | String that is utilized to define certain tags to the Node in question, a way for the user to categorize the numerous Nodes in the cluster.                                                 |
 | Annotations          | Definitions that can describe characteristics of the Node that later on can be used to process/group the Nodes in your cluster.                                                             |
 | parent               | Defines the Node context in relation to the cluster for example `app1.app2` would mean that this Node is located on the path `root->app1->app2`. It is injected by the Inspr daemon.        |
-| sha256               | tags images with their sha256 digest.                                                                                                                                                       |
+| uuid                 | Universally unique identifier for the Node.                                                                                                                                                 |
```
Comment (franciscobonand) [inline docs/yamls/dapp.md]: As this is the YAML definition for dApps/Nodes, I believe there's no need for UUID here as the user will never set it's value

File: docs/yamls/dapp.md [hunk 10]
```diff
@@ -11,25 +11,31 @@
...
 | &rarr; &rarr; Input  | List of Channels that are used for the input of this dApp                                                                                                                                   |
 | &rarr; &rarr; Output | List of Channels that are used as the output of this dApp                                                                                                                                   |
+| SidecarPort          | Strucuter that specifies the ports used to talk with the load balancer                                                                                                                      |
+| &rarr; LBRead        | The port in which the load balancer's message will be read from in the node                                                                                                                 |
+| &rarr; LBWrite       | The port in which the node will send messages to communicate with the loadbalancer                                                                                                          |
```
Comment (franciscobonand) [inline docs/yamls/dapp.md]: Shouldn't be capitalized (as it's not going to be capitalized in the YAML file)

File: docs/yamls/dapp.md [hunk 11]
```diff
@@ -11,25 +11,31 @@
...
 | &rarr;Node           | Section describing the dApp Node                                                                                                                                                            |
 | &rarr;&rarr;Meta     | Metadata of the Node                                                                                                                                                                        |
 | name                 | Defines Node name                                                                                                                                                                           |
 | reference            | String that is utilized to define certain tags to the Node in question, a way for the user to categorize the numerous Nodes in the cluster.                                                 |
 | Annotations          | Definitions that can describe characteristics of the Node that later on can be used to process/group the Nodes in your cluster.                                                             |
```
Comment (franciscobonand) [inline docs/yamls/dapp.md]: Shouldn't be capitalized (as it's not going to be capitalized in the YAML file)

File: docs/yamls/type.md [hunk 12]
```diff
@@ -2,19 +2,19 @@
...
+| apiVersion        | Specify what version of the API to use, for example `"v1"`                                                                                                         |
+| kind              | Specifies which structure the file represents, in this case it would be `type`                                                                                     |
+| meta              | Metadata of Type                                                                                                                                                   |
+| &rarr;name        | Type's Name                                                                                                                                                        |
 | &rarr;reference   | String that is utilized to define certain tags to the Type in question, a way for the user to categorize the numerous Types in the cluster.                        |
 | &rarr;Annotations | Definitions that can describe characteristics of the Type that later on can be used to process/group the Types in your cluster.                                    |
```
Comment (franciscobonand) [inline docs/yamls/type.md]: Shouldn't be capitalized (as it's not going to be capitalized in the YAML file)

File: docs/yamls/type.md [hunk 13]
```diff
@@ -2,19 +2,19 @@
...
 | &rarr;parent      | Defines the Type context in the cluster through the path of the dApp in which it is stored, for example: `app1.app2` means that the Type is defined in the `app2`. |
 |                   |
-| &rarr;sha256      | Tags images with their sha256 digest.                                                                                                                                              |
-| schema            | defines the data structure that goes through this Type, example:  `'{"type":"int"}'`                                                                                       |
-| connectedchannels | Is a list of Channels names that are created using this specific type.                                                                                                             |
+| &rarr;uuid        | Universally unique identifier for the Type.                                                                                                                        |
```
Comment (franciscobonand) [inline docs/yamls/type.md]: Same thing as commented before for the other UUID fields

**Review by franciscobonand — CHANGES_REQUESTED** [review]
**In the Channels file:**
- you can remove the double quotes from "v1" in the apiVersion field
- the "reference" field description is wrong
- the brokerlist is missing from the Definitions and from the YAML example
- "connectedaliases" is misspelled  

**In the dApps file:**
- reference field description is wrong
- you could create a better YAML example (and channel's names shouldn't have "_")

**In the Types file:**
- you can remove the double quotes from "v1" in the apiVersion field
- reference field description is wrong
- "connectedchannels" description could be better

**In index.md**
- not all channels create a new message broker topic
- in the "Types" description it says the schema should be an Avro Schema, but then proceeds to say "a string containing a json structure". This is not right
- Aliases are used to connect multi-layered dApps through channels, and not just rename the Channels. The name actually could remain the same
- In "General file" the beginning of the sentence is "For example a basic example". This is odd
- The example itself you can leave for me to update it once I finish fixing the `primes` example

File: docs/yamls/alias.md (hunk 6 above)
Comment (nicholaspcr) [reply docs/yamls/alias.md]: i'll admit that the convention is not fresh in my mind, will look in how it should be done

Comment (franciscobonand) [reply docs/yamls/alias.md]: I can help you with that

File: docs/yamls/alias.md [hunk 14]
```diff
@@ -0,0 +1,28 @@
+## Alias 
+
+### Definitions
+
+| Field             | Meaning                                                                                                                                                                                                      |
+| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| apiVersion        | Specify what version of the API to use, for example `"v1"`                                                                                                                                                   |
+| kind              | Specifies which structure the file represents, in this case it would be `alias`                                                                                                                              |
+| meta              | Metadata of Type                                                                                                                                                                                             |
```
Comment (franciscobonand) [inline docs/yamls/alias.md]: "Metadata of Type"

**Review by franciscobonand — CHANGES_REQUESTED** [review]
### Type

- apiVersion => "v1"

- reference => update to a definition that talks about registry

### Alias

- apiVersion => "v1"
- reference => update to a definition that talks about registry
- YAML example => "name" is still "aliasOne"

### Channels

- apiVersion => "v1"
- reference => update to a definition that talks about registry
- spec.type => description kinda confusing
- YAML example => update it (remove/fix annotations and add "brokerlist" field)

### dApps

- reference => update to a definition that talks about registry
- "Spec" => field starting with capital letter

### Index

- In "Types" there is still "This would be either a string containing a **json structure **"
- Aliases are used to connect multi-layered dApps through channels, and not just rename the Channels. The name actually could remain the same
- In "General file" the beginning of the sentence is "For example a basic example". This is odd
- The example itself you can leave for me to update it once I finish fixing the `primes` example

**Review by franciscobonand — APPROVED** [review]

**Review by LucasStarlingdePaulaSalles — APPROVED** [review]

**Pull request merged**